lib/xogmios/health_check.ex

```diff
@@ -0,0 +1,68 @@
+defmodule Xogmios.HealthCheck do
+  @moduledoc """
+  Performs a health check against the Ogmios server's HTTP endpoint.
+  Used primarily to determine if the underlying Cardano node is fully synced.
+  """
+
+  @http_client :httpc
+
+  @spec run(url :: String.t()) :: :ok | {:error, String.t()}
+  def run(ogmios_url) do
+    url = parse_url(ogmios_url)
+
+    # NOTE: Investigate if there's a better way to start these.
+    :inets.start()
+    :ssl.start()
```

Comment:
The better way is to replace all this nonsense with simply using [Req](https://hex.pm/packages/req)